### ISSUE TITLE
Unification creates a cyclic term

### ISSUE THREAD

**Issue by pmoura** [terminusdb/terminus-server#100]
Using a new linter check in the current Logtalk git version, I get (in the `dev` branch):

```text
*     Suspicious call: A=literal(type(A,B)) as unification will succeed creating a cyclic term
*       while compiling object woql_compile
*       in file /Users/pmoura/terminus-server/library/woql_compile.pl between lines 612-619
```

Typo in the code?

**Comment by GavinMendelGleason** [comment]
Yeah, looks like a typo alright!

**Comment by pmoura** [comment]
Fixed in 7c2421fd89cd502afe255d6de1820c59ea474399
